app/controllers/HomePageController.scala

```diff
@@ -32,15 +33,41 @@ class HomePageController @Inject() (
   getOrCreate: DataRetrievalOrCreateAction,
   profileAuth: ProfileAuthenticateAction,
   downloadDataConnector: DownloadDataConnector,
+  goodsRecordConnector: GoodsRecordConnector,
   val controllerComponents: MessagesControllerComponents,
   view: HomePageView
 )(implicit ec: ExecutionContext)
     extends BaseController {
 
   def onPageLoad: Action[AnyContent] = (identify andThen profileAuth andThen getOrCreate).async { implicit request =>
-    downloadDataConnector.getDownloadDataSummary(request.eori).map {
-      case Some(downloadDataSummary) if downloadDataSummary.status == FileReadyUnseen => Ok(view(downloadReady = true))
-      case _                                                                          => Ok(view(downloadReady = false))
+    for {
+      downloadDataSummary <- downloadDataConnector.getDownloadDataSummary(request.eori)
+      goodsRecords        <- goodsRecordConnector.getRecords(request.eori, 1, 1)
+      doesGoodsRecordExist = goodsRecords.exists(_.goodsItemRecords.nonEmpty)
+    } yield {
+      val downloadLinkMessagesKey = getDownloadLinkMessagesKey(downloadDataSummary, doesGoodsRecordExist)
+      downloadDataSummary match {
+        case Some(downloadDataSummary) if downloadDataSummary.status == FileReadyUnseen =>
+          Ok(view(downloadReady = true, downloadLinkMessagesKey))
+        case _                                                                          =>
+          Ok(view(downloadReady = false, downloadLinkMessagesKey))
+      }
     }
   }
+
```

Comment:
AC 2.2 & 2.3 ✅: If user has goods records in their profile display hyperlink, If user does not have goods records in their profile display -> Logic seems fine